File: examples/jte-example/src/main/java/de/tschuehly/example/jte/web/layout/LayoutViewComponent.java
package de.tschuehly.example.jte.web.layout;

import de.tschuehly.spring.viewcomponent.core.IViewContext;
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;

@ViewComponent
public
class LayoutViewComponent {

    private record LayoutView(IViewContext nestedViewComponent) implements IViewContext{}

    public IViewContext render(IViewContext nestedViewComponent) {
        return new LayoutView(nestedViewComponent);
    }
}

File: examples/jte-example/src/main/kotlin/de/tschuehly/jteviewcomponentdemo/web/layout/LayoutViewComponent.kt
package de.tschuehly.jteviewcomponentdemo.web.layout

import de.tschuehly.spring.viewcomponent.core.IViewContext
import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.jte.ViewContext

@ViewComponent
class LayoutViewComponent {
    data class LayoutView(val nestedViewComponent: ViewContext) : ViewContext

    fun render(nestedViewComponent: ViewContext) = LayoutView(nestedViewComponent)

}